File: src/Models/Sale.hs
module Models.Sale where
import Data.Time.Clock (UTCTime)
import Models.Product (Product)

data Sale = Sale
  { clientId   :: Int
  , sellerId   :: Int      
  , totalSale  :: Double
  , dateSale   :: UTCTime
  , products   :: [Product] 
  } deriving (Show, Read, Eq)

File: prolog/src/models/Sale.pl
:- module(sale, [sale/5, show_sale/1]).

sale(ClientId, SellerId, DateSale, TotalSale, Products).

show_sale(sale(ClientId, SellerId, DateSale, TotalSale, Products)) :-
    format('Client ID: ~d~n', [ClientId]),
    format('Seller ID: ~d~n', [SellerId]),
    format('Date of Sale: ~w~n', [DateSale]),
    format('Total Sale: ~2f~n', [TotalSale]),
    format('Products: ~w~n', [Products]).
